contracts/Remittance.sol

```diff
@@ -13,11 +13,11 @@ contract Remittance is Pausable {
   constructor(bool paused) Pausable(paused) public {
   }
 
-  function calcuateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
+  function generateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
      return keccak256(abi.encodePacked(this, _carol, _bobPassword, _carolPassword));
   }
 
-  function deposit(bytes32 _hash, address _carol) external payable whenRunning whenAlive {
+  function deposit(bytes32 _hash, address _carol) external payable onlyOwner whenRunning whenAlive {
```

Comment:
Still does not protect the user from being played by the owner.

Reply:
If I understand correctly, the owner can `deposit` twice. For example.
```
// First deposit
const hash1 = remittance.methods.generateHash('password1', 'password2', address1).call();
deposit(hash1, address1); // deposit 10 ethers

// Second deposit
const hash2 = remittance.methods.generateHash('different-password3', 'different-password4', address2).call();
deposit(hash2, address2); // deposit 5 ethers
```

Then `address2` can withdraw `5 ethers and `address1` lost all 10 ethers.

Is it the hack you are thinking?

Thanks again

Reply:
Yes. I am thinking of this. Not really a hack, more like an ability for the owner to shoot itself in the foot.

Reply:
thanks, I have fixed that in the PR #3

Reply:
thanks, I have fixed that in the PR #3